src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -171,6 +172,11 @@ typedef struct {
     uint64_t win_id;
     struct MPIDI_OFI_win_request *syncQ;
     MPIDI_OFI_win_targetinfo_t *winfo;
+
+    /* Accumulate related info. The struct internally uses MPIDI_OFI_DT_SIZES
+     * defined in ofi_types.h to allocate the max_count array. The struct
+     * size is unknown when we load ofi_pre.h, thus we only set a pointer here. */
+    struct MPIDI_OFI_win_acc_hint *acc_hint;
```

Comment:
Yes. `same_op_no_op` is the default in MPICH 3.1.

We plan to also support `any_op`, which allows the user program to call `MPI_Accumulate` with different ops. It is also being proposed to MPI-4 as the default.

Reply:
Okay. I thought it would be helpful to comment that this was only used for `same_op_no_op` case, because the current name `acc_hint` is too generic and gives an impression that this is always used.

Reply:
We may always need to malloc `acc_hint` and prepare the hint info no matter which value `accumulate_ops` is. Because `accumulate_ops` may be set at win creation time and then changed by a `win_set_info`.

We may prepare `acc_hint` only when user sets to `same_op_no_op` at win creation, but it then requires a hook to allow OFI to check the change and prepare `acc_hint` at win_set_info time. It might be easier for us to always initialize `acc_hint` and think about restriction when the info hook design is decided.